app/controllers/admin/items_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Admin::ItemsController < AdminController
+
+  def index
+    if current_user.nil? || current_user.user?
+      redirect_to root_path, alert: "Sensei says: 'Not quite yet young grasshopper'"
+    else
+      @items = Item.all
+    end
+  end
+
+  def edit
+    if current_user.nil? || current_user.user?
+      redirect_to root_path, alert: "Sensei says: 'Not quite yet young grasshopper'"
+    else
+      render :edit
+    end
+  end
+
+  def new
+    if current_user.nil? || current_user.user?
+      redirect_to root_path, alert: "Sensei says: 'Not quite yet young grasshopper'"
+    else
+      @item = Item.new
+      @categories = Category.all
+    end
+  end
+
+  def create
+    if current_user.nil? || current_user.user?
+      redirect_to root_path, alert: "Sensei says: 'Not quite yet young grasshopper'"
+    else
+      @item = Item.new(item_params)
+
+      @categories = params.select{ |k,v| k.include?("ItemsCategory") }.values[0].select{ |k,v| v == "1" }.keys
```

Comment:
Unused block argument - `v`. If it's necessary, use `_` or `_v` as an argument name to indicate that it won't be used.<br>Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Line is too long. [110/80]<br>Space missing after comma.<br>Space missing to the left of {.